### ISSUE TITLE
Issue in NextJS 12

### ISSUE THREAD

**Issue by curious-toy** [pmndrs/react-xr#118]
Hi

this code in index.js

```
import { useRef, useState } from 'react'
import { Canvas, useFrame } from '@react-three/fiber'
import { VRCanvas } from '@react-three/xr'

function Box(props) {
  // This reference gives us direct access to the THREE.Mesh object
  const ref = useRef()
  // Hold state for hovered and clicked events
  const [hovered, hover] = useState(false)
  const [clicked, click] = useState(false)
  // Subscribe this component to the render-loop, rotate the mesh every frame
  useFrame((state, delta) => (ref.current.rotation.x += 0.01))
  // Return the view, these are regular Threejs elements expressed in JSX
  return (
    <mesh
      {...props}
      ref={ref}
      scale={clicked ? 1.5 : 1}
      onClick={(event) => click(!clicked)}
      onPointerOver={(event) => hover(true)}
      onPointerOut={(event) => hover(false)}>
      <boxGeometry args={[1, 1, 1]} />
      <meshStandardMaterial color={hovered ? 'hotpink' : 'orange'} />
    </mesh>
  )
}

export default ()=>{

  return (
    <VRCanvas>
      <ambientLight />
      <pointLight position={[10, 10, 10]} />
      <Box position={[-1.2, 0, 0]} />
      <Box position={[1.2, 0, 0]} />
    </VRCanvas>
  )

}
```
produces this console error:

```
Uncaught     at C:\Users\61420\curious_toy\website\node_modules\ (webxr-input-profiles\motion-controllers\dist\motion-controllers.module.js:397)
    at wrapSafe (internal/modules/cjs/loader.js:988:16)
    at Module._compile (internal/modules/cjs/loader.js:1036:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (file://C:\Users\61420\curious_toy\website\node_modules\@react-three\xr\dist\index.cjs.js:10:25)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
```

**Comment by curious-toy** [comment]
Hi @CodyJasonBennett 

I upgraded and hit this this console error

```
SyntaxError: Cannot use import statement outside a module

Uncaught     at <unknown> (file://C:\Users\61420\curious_toy\website\node_modules\three-stdlib\loaders\GLTFLoader.js:1)
    at wrapSafe (internal/modules/cjs/loader.js:988:16)
    at Module._compile (internal/modules/cjs/loader.js:1036:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (file://C:\Users\61420\curious_toy\website\node_modules\@react-three\xr\dist\index.cjs.js:10:18)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
```

**Comment by CodyJasonBennett** [comment]
Can you create a reproduction? This is reliant on your config.

**Comment by curious-toy** [comment]
Hi @CodyJasonBennett 

yarn create next-app
yarn add three @react-three/fiber @react-three/xr

pages/index.js:
```
import { VRCanvas } from '@react-three/xr'
import React, { useRef, useState } from 'react'
import { useFrame } from '@react-three/fiber'

function Box(props) {
  // This reference gives us direct access to the THREE.Mesh object
  const ref = useRef()
  // Hold state for hovered and clicked events
  const [hovered, hover] = useState(false)
  const [clicked, click] = useState(false)
  // Subscribe this component to the render-loop, rotate the mesh every frame
  useFrame((state, delta) => (ref.current.rotation.x += 0.01))
  // Return the view, these are regular Threejs elements expressed in JSX
  return (
    <mesh
      {...props}
      ref={ref}
      scale={clicked ? 1.5 : 1}
      onClick={(event) => click(!clicked)}
      onPointerOver={(event) => hover(true)}
      onPointerOut={(event) => hover(false)}>
      <boxGeometry args={[1, 1, 1]} />
      <meshStandardMaterial color={hovered ? 'hotpink' : 'orange'} />
    </mesh>
  )
}

export default ()=>{

  return (
    <VRCanvas>
      <ambientLight />
      <pointLight position={[10, 10, 10]} />
      <Box position={[-1.2, 0, 0]} />
      <Box position={[1.2, 0, 0]} />    
    </VRCanvas>
  )

}
```

**Comment by curious-toy** [comment]
Hi @CodyJasonBennett 

Does this commit fix the issue?

I switched out the lines of code you committed and I still get the error.

**Comment by CodyJasonBennett** [comment]
@curious-toy, it should yes. We really need to get a publishing pipeline going but you can try it at `@codyjasonbennett/xr@3.4.1`.

**Comment by verekia** [comment]
I confirm that `@codyjasonbennett/xr@3.4.1` solved the issue for me :) Great work.
Any idea when we could get an official release? Thank you.

**Comment by CodyJasonBennett** [comment]
I'll have to cc @sniok on this.

**Comment by curious-toy** [comment]
Sorry for late reply, have been away. Thanks for the fix. Looking forward to official release. 👍

**Comment by CodyJasonBennett** [comment]
Fix published in `3.5.0`.
